api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -112,9 +114,25 @@
     <T> T getValue(String propertyName, Class<T> propertyType);
 
     /**
-     * Return the resolved property values with the specified type for the
-     * specified property name from the underlying {@linkplain ConfigSource configuration sources}.
+     * Return the {@link ConfigValue} for the specified property name from the underlying
+     * {@linkplain ConfigValueSource configuration sources}.
      * <p>
+     *
+     * The configuration value is not guaranteed to be cached by the implementation, and may be expensive
+     * to compute; therefore, if the returned value is intended to be frequently used, callers should consider storing
+     * rather than recomputing it.
+     *
+     * @param propertyName
+     *              The configuration property name
+     * @return the ConfigValue, or {@code null} if the property is not present
```

Comment:
Should null be returned if there is no ConfigValueSource supplied?

Reply:
Presumably all ConfigSources would be wrapped, so no.